### PR TITLE
Java: Automodel Extraction Parameter Name Fix

### PR DESCRIPTION
**Pull request by kaeluka**
The exported `name` metadata parameter used to wrongly contain the _parameter_ name, but should've contained the _method_ name.

This fixes the bug, and exports the parameter name as `parameterName` metadata parameter.

### CODE REVIEW

File: java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll
Comment (tausbn) [inline java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll:178]:
```suggestion
      parameterName = e.asParameter().getName() and
```
I don't know for sure that this is the right way to express this, but we should be using something like this instead of `toString`.

Comment (kaeluka) [reply java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll:178]: OMG, I looked for this but didn't find it. Thx!

**Review by tausbn — APPROVED** [review]

**Comment by kaeluka** [comment]
I'll wait for the experiment to conclude before I merge this 👍

File: java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll
Comment (atorralba) [inline java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll:178]: Note that this forces `e` to always be a `Parameter`, and `hasMetadata` will not hold if it's anything else. I guess this is fine in framework mode, but I just wanted to mention it.

**Review by atorralba — APPROVED** [review]
LGTM, one minor comment to keep in mind (probably no action required).

File: java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll
Comment (tausbn) [reply java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll:178]: This is a good point. Perhaps instead we should add a `getName` method to the interface for `EndPoint`, and call that. Internally (for the version for which  `EndPoint` is `ParameterNode`) we can then call `.asParameter().getName()`.

Comment (kaeluka) [reply java/ql/src/Telemetry/AutomodelFrameworkModeCharacteristics.qll:178]: > I guess this is fine in framework mode, but I just wanted to mention it.

I think that's completely fine here! In application mode, we wouldn't export the method parameter at all (it's not available there)

**Comment by kaeluka** [comment]
The experiment came back; the fix is working, but I forgot to add the extra `$@`s in the message string, so they're missing from the sarif 🙈 fixed now.

**Review by atorralba — APPROVED** [review]

**Pull request merged**
